packages/runtime-dom/src/components/Transition.ts

```diff
@@ -194,7 +194,7 @@ export function resolveTransitionProps(
       callHook(hook, [el, resolve])
       nextFrame(() => {
         removeTransitionClass(el, isAppear ? appearFromClass : enterFromClass)
-        if (__COMPAT__ && legacyClassEnabled) {
+        if (__COMPAT__ && legacyClassEnabled && legacyAppearFromClass && legacyEnterFromClass) {
```

Comment:
@SxDx  I think a more straightforward solution could be to initialize these legacy*Class variables as empty strings on line 143-145